Lib/typing.py

```diff
@@ -2412,10 +2414,81 @@ def utf8(value: bytes) -> bytes: ...
       def utf8(value: str) -> bytes: ...
       def utf8(value):
           # implementation goes here
+
+    The overloads for a function can be retrieved at runtime using the
+    get_overloads() function.
     """
+    # Inline version of _get_key_for_callable
+    # classmethod and staticmethod
+    inner_func = getattr(func, "__func__", func)
+    try:
+        key = (inner_func.__module__, inner_func.__qualname__)
+    except AttributeError:
+        # Not a normal function; ignore.
+        pass
+    else:
+        # We're inlining get_overloads() here to avoid computing the key twice.
+        existing = _overload_registry.setdefault(key, [])
+        if existing:
+            # If we are registering a variant with a lineno below or equal to that of the
+            # most recent existing variant, we're probably re-creating overloads for a
+            # function that already exists. In that case, we clear the existing variants
+            # to avoid leaking memory.
+            try:
+                firstlineno = inner_func.__code__.co_firstlineno
+            except AttributeError:
+                pass
+            else:
+                existing_func = existing[-1]
+                # classmethod and staticmethod
+                existing_func = getattr(existing_func, "__func__", existing_func)
+                try:
+                    existing_lineno = existing_func.__code__.co_firstlineno
+                except AttributeError:
+                    pass
+                else:
+                    if firstlineno <= existing_lineno:
+                        existing.clear()
+
+        existing.append(func)
     return _overload_dummy
 
 
+# {key: [overload]}
+_overload_registry: dict[tuple[str, str], list[Any]] = {}
```

Comment:
```suggestion
_overload_registry: collections.defaultdict[tuple[str, str], list[Any]] = collections.defaultdict(list)
```

Reply:
Thanks, applied this manually (dumped the type annotation since we're not checking it and it's getting on the long side)